zerver/webhooks/airbyte/view.py

```diff
@@ -0,0 +1,95 @@
+# Webhooks for external integrations.
+
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import webhook_view
+from zerver.lib.response import json_success
+from zerver.lib.typed_endpoint import JsonBodyPayload, typed_endpoint
+from zerver.lib.validator import WildValue, check_bool, check_int, check_string
+from zerver.lib.webhooks.common import check_send_webhook_message
+from zerver.models import UserProfile
+
+AIRBYTE_TOPIC_TEMPLATE = "{workspace} - {connection}"
+
+AIRBYTE_MESSAGE_TEMPLATE = """
+{sync_status_emoji} Airbyte sync **{status}** for [{connection_name}]({connection_url}).
+
+
+* **Source:** [{source_name}]({source_url})
+* **Destination:** [{destination_name}]({destination_url})
+* **Records:** {records_emitted} emitted, {records_committed} committed
+* **Bytes:** {bytes_emitted_formatted} emitted, {bytes_committed_formatted} committed
+* **Duration:** {duration_formatted}
+"""
+
+
+def extract_data_from_payload(payload_data: WildValue) -> dict[str, str | int | bool]:
+    return {
+        "workspace_name": payload_data["workspace"]["name"].tame(check_string),
+        "connection_name": payload_data["connection"]["name"].tame(check_string),
+        "source_name": payload_data["source"]["name"].tame(check_string),
+        "destination_name": payload_data["destination"]["name"].tame(check_string),
+        "connection_url": payload_data["connection"]["url"].tame(check_string),
+        "source_url": payload_data["source"]["url"].tame(check_string),
+        "destination_url": payload_data["destination"]["url"].tame(check_string),
+        "successful_sync": payload_data["success"].tame(check_bool),
+        "duration_formatted": payload_data["durationFormatted"].tame(check_string),
+        "records_emitted": payload_data["recordsEmitted"].tame(check_int),
+        "records_committed": payload_data["recordsCommitted"].tame(check_int),
+        "bytes_emitted_formatted": payload_data["bytesEmittedFormatted"].tame(check_string),
+        "bytes_committed_formatted": payload_data["bytesCommittedFormatted"].tame(check_string),
+        "error_message": payload_data.get("errorMessage").tame(check_string)
+        if payload_data.get("errorMessage")
+        else "No error",
```

Comment:
My understanding is that `payload_data["errorMessage"]` always exists but is `None` when no errors occur. When we use `payload_data.get("errorMessage")`, it returns a falsy value in these cases. We then replace this with “No error,” but this value is never actually used afterward. This results in unnecessarily complicated logic IMO.